Meadowlark-js/packages/meadowlark-core/src/extraction/BuildDocumentInfo.ts

```diff
@@ -2,35 +2,37 @@
 // Licensed to the Ed-Fi Alliance under one or more agreements.
 // The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
 // See the LICENSE and NOTICES files in the project root for more information.
-import { TopLevelEntity } from '@edfi/metaed-core';
-import { extractDocumentReferences } from './DocumentReferenceExtractor';
-import { extractDocumentIdentity, deriveSuperclassInfoFrom } from './DocumentIdentityExtractor';
+
 import { DocumentInfo } from '../model/DocumentInfo';
 import { extractDescriptorValues } from './DescriptorValueExtractor';
-import { SuperclassInfo } from '../model/SuperclassInfo';
+import { extractDocumentIdentity } from './DocumentIdentityExtractor';
+import { extractDocumentReferences } from './DocumentReferenceExtractor';
+import { deriveSuperclassInfoFrom } from './SuperclassInfoExtractor';
 import { DocumentIdentity } from '../model/DocumentIdentity';
-import { ResourceInfo } from '../model/ResourceInfo';
+import { SuperclassInfo } from '../model/SuperclassInfo';
+import { ApiSchema } from '../model/api-schema/ApiSchema';
+import { ResourceSchema } from '../model/api-schema/ResourceSchema';
 
 /**
- * Extracts document identity and document reference information from the request body.
+ * Builds a DocumentInfo using the various extractors
  */
-export async function extractDocumentInfo(
-  resourceInfo: ResourceInfo,
+export function buildDocumentInfo(
+  apiSchema: ApiSchema,
+  resourceSchema: ResourceSchema,
   body: object,
-  matchingMetaEdModel: TopLevelEntity,
   requestTimestamp: number,
-): Promise<DocumentInfo> {
-  const documentIdentity: DocumentIdentity = extractDocumentIdentity(matchingMetaEdModel, body);
+): DocumentInfo {
+  const documentIdentity: DocumentIdentity = extractDocumentIdentity(resourceSchema, body);
```

Comment:
All the extractors are rewritten, probably easier to look at them directly rather than try to follow the diff